tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -474,15 +474,18 @@ static std::function<bool(PendingExecutionResult)> FinishedCondition(PendingQuer
 unique_ptr<QueryResult> DuckDBPyConnection::CompletePendingQuery(PendingQueryResult &pending_query) {
 	PendingExecutionResult execution_result;
 	auto is_finished = FinishedCondition(pending_query);
-	do {
-		execution_result = pending_query.ExecuteTask();
+	while (!is_finished(execution_result = pending_query.ExecuteTask())) {
 		{
 			py::gil_scoped_acquire gil;
 			if (PyErr_CheckSignals() != 0) {
 				throw std::runtime_error("Query interrupted");
 			}
 		}
-	} while (!is_finished(execution_result));
+		if (execution_result == PendingExecutionResult::BLOCKED ||
```

Comment:
Should this block with `NO_TASKS_AVAILABLE` as well?

Reply:
Now that you mention it, probably not?
I figured since there was nothing to do it makes no sense to busy spin when all remaining tasks are still being processed, but besides spurious wake ups or our 20ms timeout there is nothing to wake up the thread, so that's a little problematic

Reply:
I would say we shouldn't rely on the timeout - let's just keep busy spinning unless we are blocked.